### PR TITLE
Add `on:change` event, remove two-way data binding

### PR DESCRIPTION
**Pull request by dstotijn**
This PR:
* Removes the _two way_ data bind of `data`; it's just one way now. From parent component to `GOBLBuilder`.
* Adds a new `change` event that's dispatched whenever the editor content changes or the envelope is changed.

### CODE REVIEW

File: src/lib/GOBLBuilder.svelte [hunk 1]
```diff
@@ -30,7 +34,7 @@
         $editor = JSON.stringify(parsedValue.doc, null, 4);
         $envelope = parsedValue;
       } else {
-        $editor = JSON.stringify(parsedValue, null, 4);
+        $editor = data;
```
Comment (dstotijn) [inline src/lib/GOBLBuilder.svelte:37]: This change is unrelated to the purpose/goal of this PR. It just ensures that we use the `data` from the parent component "as-is" when it's not wrapped in an envelope. This allows the parent project to set the editor with invalid/incomplete JSON in case it want to.

File: src/lib/GOBLBuilder.svelte [hunk 2]
```diff
@@ -1,17 +1,21 @@
...
   export let jsonSchemaURL = "";
 
-  // Data is used for setting editor contents. It allows for two way component
-  // binding. E.g. when the editor contents are changed, this property is also
-  // updated.
+  // Data is used for setting editor contents. Note: there is "one way" binding;
+  // e.g. you can set data but changes are not bound to the parent. Use the
+  // `change` event, to receive changes to the editor contents and GOBL
+  // envelope.
   export let data = "";
```
Comment (agorer) [inline src/lib/GOBLBuilder.svelte:19]: I would probably change the name now to show that it is only the initial data loaded on the editor.

File: src/routes/+page.svelte [hunk 3]
```diff
@@ -33,6 +33,9 @@
       bind:data
       bind:problems
       {jsonSchemaURL}
+      on:change={(event) => {
+        console.log("Received change event.", event.detail);
```
Comment (agorer) [inline src/routes/+page.svelte:37]: Playing with it in Netlify I see a couple of properties returned by the event (`editor` and `envelope`) and the envelope property is returned as _null_ sometimes. Will it return not-null as long as the document is valid or has the user to build the document to create the envelope part?

**Review by agorer — APPROVED** [review]

File: src/lib/GOBLBuilder.svelte (hunk 2 above)
Comment (dstotijn) [reply src/lib/GOBLBuilder.svelte:19]: You can update the data over time though; so it's technically not only for setting initial data, but also updating it. E.g. on the demo on https://build.gobl.org the initial data is empty, and then when you pick a template we're updating the value with a JSON string.

File: src/routes/+page.svelte (hunk 3 above)
Comment (dstotijn) [reply src/routes/+page.svelte:37]: > Will it return not-null as long as the document is valid or has the user to build the document to create the envelope part?

The `envelope` property is non-null when there's an envelope (either from the data passed in from the parent component, or the result of build or validate actions), **and** the editor content is valid JSON.
